wp-includes/class-wp-service-workers.php

```diff
@@ -83,6 +220,170 @@ public function register( $handle, $src, $deps = array(), $scope = self::SCOPE_A
 		return parent::add( $handle, $src, $deps, false, compact( 'scope' ) );
 	}
 
+	/**
+	 * Register route and caching strategy.
+	 *
+	 * @todo Figure out a better name for $route since it can also be a list.
+	 * @param string|array $route Route or list of routes.
+	 * @param string       $strategy Strategy, can be WP_Service_Workers::STRATEGY_STALE_WHILE_REVALIDATE, WP_Service_Workers::STRATEGY_CACHE_FIRST,
+	 *                         WP_Service_Workers::STRATEGY_NETWORK_FIRST, WP_Service_Workers::STRATEGY_CACHE_ONLY,
+	 *                         WP_Service_Workers::STRATEGY_NETWORK_ONLY, WP_Service_Workers::::STRATEGY_PRECACHE.
+	 * @param array        $args Array of args, can include cache_name, max_age, max_entries.
+	 */
+	public function register_cached_route( $route, $strategy, $args = array() ) {
+
+		if ( ! in_array( $strategy, array(
+			self::STRATEGY_STALE_WHILE_REVALIDATE,
+			self::STRATEGY_CACHE_FIRST,
+			self::STRATEGY_CACHE_ONLY,
+			self::STRATEGY_NETWORK_FIRST,
+			self::STRATEGY_NETWORK_ONLY,
+			self::STRATEGY_PRECACHE,
+		), true ) ) {
+			_doing_it_wrong( __METHOD__, esc_html__( 'Strategy must be either WP_Service_Workers::STRATEGY_STALE_WHILE_REVALIDATE, WP_Service_Workers::STRATEGY_CACHE_FIRST, WP_Service_Workers::STRATEGY_NETWORK_FIRST, WP_Service_Workers::STRATEGY_CACHE_ONLY, or WP_Service_Workers::STRATEGY_NETWORK_ONLY.', 'pwa' ), '0.2' );
+			return;
+		}
+
+		if ( self::STRATEGY_PRECACHE === $strategy ) {
```

Comment:
Instead of including `precache` in this method, I think it should probably be a separate method like `register_precached_route`. This then can be a straight wrapper for `workbox.precaching.precache()`: https://developers.google.com/web/tools/workbox/reference-docs/latest/workbox.precaching#.precache